pyproject.toml

```diff
@@ -193,6 +193,7 @@ files = ["src/trio/", "docs/source/*.py"]
 ignore_missing_imports = true
 
 # Be strict about use of Mypy
+enable_error_code = ["truthy-bool", "mutable-override"]
 local_partial_types = true
```

Comment:
ugh, so messy that this is a separate way of enabling checks vs having `warn_truthy_bool` (and I did a quick check for it).
also sad that `strict=True` leaves so much disabled. The workflow of `--enable=ALL` into selectively disabling is so much better.

aanyway this PR in itself seems fine